src/components/app-player/script.js

```diff
@@ -1,8 +1,158 @@
+    },
+    /**
+     * Return the song information (artist - album) if is possible
+     */
+    getSongInfo() {
+      if (this.currentTrack && this.currentTrack.trackId !== TRACK_INFO.trackId) {
+        return `${this.currentTrack.artistName} - ${this.currentTrack.collectionName}`;
+      }
+      return '';
+    },
+    /**
+     * Return true if the current trac info allow user to play something, else false.
+     */
+    isPlayEnabled() {
+      return this.currentTrack.previewUrl;
+    },
+  },
+  mounted() {
+    this.$root.$on(MESSAGE_PLAY_TRACK, this.setPlayTrack);
+    this.$refs.player.addEventListener('play', this.onReproductionStart);
+    this.$refs.player.addEventListener('pause', this.onReproductionPause);
+    this.$refs.player.addEventListener('ended', this.onReproductionEnds);
+    this.$refs.player.addEventListener('canplay', this.onTrackReady);
+    this.$refs.player.addEventListener('timeupdate', this.timeUpdate);
+    window.audio1 = this.$refs.player;
+  },
+  methods: {
+    /**
+     * Modify component's current play track.
+     * @param {number|trackInfo} trackData track identifier or track information
+     * @param {boolean} useMock define if we would use a mock
+     */
+    async setPlayTrack(trackData, useMock = false) {
+      this.$refs.player.pause();
+      if (typeof trackData === 'number') {
+        this.currentTrack = await api.lookup(trackData, useMock);
+      } else {
+        this.currentTrack = trackData;
+      }
+      this.mustPlay = true;
+    },
+
+    /**
+     * Execute proper actions when the reproduction ends.
+     */
+    onReproductionStart() {
+      this.isPlaying = true;
+      // this.startUpdateTimer();
+    },
+
+    /**
+     * Execute proper actions when the reproduction ends.
+     */
+    onReproductionEnds() {
+      this.isPlaying = false;
+      this.clearUpdateTimer();
+      this.sliderTime = 0;
+      this.currentTime = '00:00';
+    },
+
+    /**
+     * Execute proper actions when the track is paused.
+     */
+    onReproductionPause() {
+      this.isPlaying = false;
+      // this.clearUpdateTimer();
+    },
+
+    /**
+     * Execute proper actions when the track is loaded.
+     */
+    onTrackReady() {
+      this.updateVisibleTrackInfo();
+      if (this.mustPlay) {
+        this.mustPlay = false;
+        this.togglePlayTrack();
+      }
+    },
+
+    /**
+     *Modify relevant information according current track.
+     */
+    updateVisibleTrackInfo() {
+      this.duration = this.$refs.player.duration * UPDATE_INTERVAL;
+      this.sliderTime = 0;
+    },
+
+    /**
+     * Change the current play status
+     */
+    togglePlayTrack() {
+      if (!this.isPlaying) {
+        this.$refs.player.play();
+      } else {
+        this.$refs.player.pause();
+      }
+    },
+
+    /**
+     * Action to be executed on time update
+     */
+    timeUpdate() {
+      const dateTime = new Date(0);
+      dateTime.setSeconds(this.$refs.player.currentTime);
```

Comment:
I think we can grab this info from event parameter

Reply:
Done.